openedx/features/content_type_gating/block_transformers.py

```diff
@@ -68,4 +68,5 @@ def transform(self, usage_info, block_structure):
                     [settings.CONTENT_TYPE_GATE_GROUP_IDS['full_access']]
                 )
                 block_structure.override_xblock_field(block_key, 'group_access', current_access)
-                self._set_contains_gated_content_on_parents(block_structure, block_key)
+                if current_access[CONTENT_GATING_PARTITION_ID] == [settings.CONTENT_TYPE_GATE_GROUP_IDS['full_access']]:
+                    self._set_contains_gated_content_on_parents(block_structure, block_key)
```

Comment:
What's this fix about?

Reply:
The `setdefault` call earlier is because some courses explicitly set the `CONTENT_GATING_PARTITION_ID` values in `group_access`. In those case, we don't override their settings. One of the settings they can use is `[settings.CONTENT_TYPE_GATE_GROUP_IDS['full_access'], settings.CONTENT_TYPE_GATE_GROUP_IDS['limited_access']]`, to show content to both full access and limited access users. In that case, we don't want to mark that content as being gated content, even though it's eligible to be gated content.

Reply:
Was that just an unrelated bug you noticed? Good catch.

Reply:
No, that was the core bug that I was fixing.